spring/spring4/spring4-common/src/main/java/org/infinispan/spring/session/AbstractInfinispanSessionRepository.java

```diff
@@ -0,0 +1,62 @@
+package org.infinispan.spring.session;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.spring.provider.SpringCache;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.ApplicationEventPublisherAware;
+import org.springframework.session.MapSession;
+import org.springframework.session.SessionRepository;
+
+/**
+ * Infinispan implementation for Spring Session with basic functionality.
+ *
+ * @author Sebastian Łaskawiec
+ * @see <a href="http://projects.spring.io/spring-session">Spring Session Web Page</a>
+ * @see SessionRepository
+ * @see ApplicationEventPublisherAware
+ * @since 9.0
+ */
+public abstract class AbstractInfinispanSessionRepository implements SessionRepository<MapSession>, ApplicationEventPublisherAware {
+
+   protected final AbstractApplicationPublisherBridge applicationEventPublisher;
+   protected final SpringCache cache;
+   protected final PrincipalNameResolver principalNameResolver = new PrincipalNameResolver();
+
+   protected AbstractInfinispanSessionRepository(SpringCache cache, AbstractApplicationPublisherBridge eventsBridge) {
+      Objects.requireNonNull(cache, "SpringCache can not be null");
+      Objects.requireNonNull(eventsBridge, "EventBridge can not be null");
+      applicationEventPublisher = eventsBridge;
+      this.cache = cache;
+   }
+
+   @Override
+   public void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
+      this.applicationEventPublisher.setApplicationEventPublisher(applicationEventPublisher);
+   }
+
+   @Override
+   public MapSession createSession() {
+      MapSession result = new MapSession();
+      result.setCreationTime(System.currentTimeMillis());
+      return result;
+   }
+
+   @Override
+   public void save(MapSession session) {
+      cache.put(session.getId(), session, session.getMaxInactiveIntervalInSeconds(), TimeUnit.SECONDS);
```

Comment:
Hi @slaskawi , so you're using the "lifespan" on an entry and when the lifespan expires it will remove the session from the application? I'm wondering whether we should use "maxIdle" instead. When a user accesses the session very often but is just reading, the session will expire anyway even though it is accessed often. Does the user need to write to the session to keep it alive? Just wondering...